### ISSUE TITLE
Action so that Data Manager can move search links and then amend Gazetteer

### ISSUE THREAD

**Issue by geoffj-FUG** [FreeUKGen/FreeCENMigration#1566]
There are many situations where entries will need to be removed from the Gazetteer and consequently search links will need to be moved.

Is it possible to write an Action so that these changes can be made by the data manager instead of being referred to a programmer?

We have now locked down the ability to delete place names so the requests for link changes and deletions will increase.

Tried again and this time got the next screen.

Geoff

**Comment by DeniseColbert** [comment]
@Vino-S to let @AnneV-Learn know which rake task to work on for this, to give Data Manager access in application to move search links from incorrect entries in the Gazetteer.

**Comment by AnneV-Learn** [comment]
@Vino-S, which Rake task have you run when you have been asked to 'move' search records? (Apologies if you have already told me this but I can't find the answer.)

**Comment by AnneV-Learn** [comment]
Vino confirmed the existing Rake task is freecen:convert_freecen2_place_id

**Comment by AnneV-Learn** [comment]
Code nearly ready but need to wait until #1620 and #1627 have been deployed to production as there is too much crossover code.

**Comment by AnneV-Learn** [comment]
#1620 and #1627 have now been deployed but finalising the code depends on #1695

**Comment by AnneV-Learn** [comment]
@geoff-fug

Testing Move Place linkages on Test3

NOTE: Birth place does not exist on all search records on Test3 as I believe the scripts that were run on Prod to populate it for all existing search_record data were not run on Test3 by Kirk. So the way to test this new functionality is to choose data from a CSVProc file and re-load it (the CSVProc load process does populate birth_place and the link to Gazetteer place ie. Freecen2_places, for valid birth places). Also re-load a VLD file (this process also populates birth_place and the link to Gazetteer place ie. Freecen2_places, for valid birth places) and work with Birth Places used by that data.

The process (available to Data Managers and Sys Admin users only) is as follows:

	1. Click Manage Counties and choose the county for the 'Old' place. 	
	2. Click Manage FreeCEN2 Places and display the 'Old' place.
	3. Clink Move Freecen2 Place and choose the 'New' county and place (note: 'New' place must already exist in the Gazetteer)
	4. Click Review Details - and check that the places identified are the ones you intended.
	5. Initially select Review mode and click Move Place Linkages. This will set off a RAKE task and you will be notified by email when it has completed.  	
	6. Upon receiving the email review the attached log file and make sure all looks as expected.
	7. If all OK in email log repeat steps 3 -> 5 but choose Update mode.  
	Notes: 
	a. If 'alternative' place names from the Old place need to be added to the New place it will be reported in the log 				file from the Update mode run and can be done after that has run. 
	b. The 'Old' place will be ’soft' deleted by this process. 

IMPORTANT : Following successful testing this branch cannot be deployed to Production until the running of rake task to populate the FreeCen2_place_of_birth_id on Production search_records data has been run across all counties (currently ongoing) and the set up of an index on that field have been completed.

**Comment by geoffj-FUG** [comment]
Test 1
Remove SOM Backwell Voyseyville and replace it with SOM Backwell
Result whilst waiting for Review ewsults:
_Proxy Error
The proxy server received an invalid response from an upstream server.
The proxy server could not handle the request
Reason: Error reading from remote server_

Next
Backwell Churchtown exists as a place name and as an alternative place name. I selected it but could only move the links to Backwell (not the alternative existing name of Backwell Churchtown.
I got the same Proxy error.

I went back and now the breadcrumbs do not work - I wnated to go and check a vld file entryy that I had uploaded for a precious test. I am locked up (not even Your Actions works).

Closed test3 and reopened it. Just a spinner at the top of Google.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
Test1 - strange as it worked for me. 
![IMG_0557](https://github.com/FreeUKGen/FreeCENMigration/assets/72746727/1fc728db-5213-4cb2-a076-b9fc78a310b7)
![IMG_0558](https://github.com/FreeUKGen/FreeCENMigration/assets/72746727/ddd7d242-71e7-4e16-83c8-5df5df13d644)
and then I received the Email for the review with attached log
[Move_freecen2_place.Log](https://github.com/FreeUKGen/FreeCENMigration/files/14931748/Move_freecen2_place.Log)
Could you please try that one again Geoff. 
NEXT:
Is the Gazetteer entry with  the ‘Alternate name’ correct or the seperate Gazetteer entry?
If you want to move linkages for the separate entry (Backwell Churchtown) then you should choose that and then the Gazetteer entry with  the ‘Alternate name’ (Backwell) as the ‘new place’. It will move any linkages to the separete  entry - Backwell Churchtown-  replacing them with linkages to the Gazetteer entry with  the ‘Alternate name’ (Backwell) and then delete the separate entry. Which will be correct as you know that it has Backwell Churchtown as an alternative name.

**Comment by geoffj-FUG** [comment]
Anne

I am using RG11_2353 as my test piece.
Row 38. Change from Henley HAM to Abbots Worthy HAM.
Result - Waiting for Review Details. After a while got
Proxy Error
The proxy server received an invalid response from an upstream server.
The proxy server could not handle the request
Reason: Error reading from remote server

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG  Ok I was able to replicate problem this time - it is due to processing time taken to gather some of the data related to if the place is used as POB. I have removed that info from the Review  screen and it appears to run more quickly now. The info on use of place as a POB is still reported in the log file sent by email when 'Move Linkages' Rake task in Review Only and Update modes complete.

**Comment by geoffj-FUG** [comment]
Still getting
Proxy Error
The proxy server received an invalid response from an upstream server.
The proxy server could not handle the request

Retried and worked OK. Got the review report OK. Cancelled at the Review point. (these places had no records).

Now looking at moving Axbridge Somerset to backwell Voyseyville Somerset. This involves 24 files and 12503 records. All 1851.
Proxy error again.
Used back arrow and reset Backwell Voyseyville. Successful display. 12503 records for Axbridge 0 records for Voyseyville.

Review report received OK. Files listed OK. Records present as expected:
Old Place Search Data present = true
New Place Search Data present = false
Selected update. 
Old Place Search Records count = 11424
Old Place Census Search Records - CSV File(s) = false
Old Place Census Search Records - VLD File(s) = true
** Search Records Places Of Birth **
Old Place name used in Search Records as Place Of Birth count = 1083

I dont understand why the count has dropped from 12000+ to 11424. No doubt the 1083 has something to do with it.

Check - Voyseyville now has 11424 search records, 
Axbridge is now missing from the listing for SOM. Checked Gazetteer and Axbridge has been removed.
When I re-created Axbridge I am told that there are 1079 records online. These must previously have been unlinked.
I clicked on gazetter to create Axbridge and it has overwritten everything I have done so far. 
Axbridge is now in teh SOM list

Clicked on Review Details
Worked fine this time. Voyseyville shows 11424 search records and Axbridge 1079. 24 pieces shown against Voyseyville (correct).
Review rake run
Old Place Districts count = 7
** Pieces **
Old Place Pieces count = 7
** Civil Parishes **
Old Place Civil Parishes count = 9
** Search Records **
Old Place Search Records count = 11424
Old Place Census Search Records - CSV File(s) = false
Old Place Census Search Records - VLD File(s) = true
** Search Records Places Of Birth **
Old Place name used in Search Records as Place Of Birth count = 1083 (this confuses me as Axbridge count was 1079).

Update run started. Report received. Counts of 11424and 1083 remain.

Checked Axbridge in Gazetteer - reports 12503 search records, 24 files - this is where we started.
Backwell Voyseyville is no longer in Gazetteer. (In case you are wondering Voyseyville is fictitious).

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
I have made corrections to sort the problem with the counts - I think it looks ok now I 'replicated' your test. But please re-test and maybe do another different test (where the linkages are moved between counties if possible).
I also changed the screen that it returns to after running the Move Place Linkages in 'Update' mode - as it might be pretty confusing to return to the screen for the 'old' place which will in fact not exist after the Rake run in Update mode has completed. In 'Review only' mode it still returns to the 'Review' screen as normally I would expect the user to just wait on that screen until they have received and reviewed the email from the 'Review Only' mode run and the go ahead with the run in Update mode from there.
I didn't get any Proxy Errors - I think we may have to see if that happens when the code goes live - the data and indexes in the Production environment are much more reliable than Test3.

**NOTE:** we won't be able to move this code to production until after I have finished adding the POB place ids to the Search Records on the Production database. I am about 3/4 way through that now. Then we will need to add one or more indexes to support this (and eventually the POB Searching).

**Comment by geoffj-FUG** [comment]
Test objective
Move Bristol GLS search links to Backwell Hill SOM and back again.

Bristol - 18286 search records, 275 pieces, years 1861 abd 1891
Backwell Hill - no records on line
Old Place Search Records count = 18286
Old Place Census Search Records - CSV File(s) = false
Old Place Census Search Records - VLD File(s) = false
** Search Records Places Of Birth **
Old Place name used in Search Records as Place Of Birth count = 6054
Old Place as Place of Birth is used in the following CSVProc Files [(SOM) HO107_1943.csv, (BRE) RG14_33563.csv] - these files MAY need to be updated and reloaded after the linkages move has been run in UPDATE mode
Old Place as Place of Birth is used in the following VLD Files [(CON) ho511909.vld,

So this change will affect both vld and csv files (the Axbridge test only affected vld files).
Still not sure about the meaning of the 6064 count.
Ran update - Backwell Hill now has 18286 records inn 275 files. Bristol GLS has been removed from the Gazetteer but Bristol SOM has not (correct). All other Bristol entries are there.
Bristol entered into Gazetteer 
Now Backwell Hill selected to move from 18286 search records in 275 files. Move 2to selected as Bristol GLS, Bristol shown as no search records.
Review report same as before.
Update run.
18286 records moved.
Backwell Hill has been removed. Bristol has 18286 search records in 275 files.

Comment
When I run the review it is taking over my Gazetteer tab instead of appearing in my FreeCEN tab. So I end up with no gazetteer and 2 FreeCEN tabs.

Question
Axbridge in particular is a link to the PARMS but it was deleted from the Gazetteer in the previous test. How was that link handled? (Bristol probably is too but I will be specific with Axbridge).

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG can you please forward me the emails you received for the tests you have done  ie 2 for each  test  (Review & Update). I’ll then annotate to better explain the output and what is going on. 
The code is based on a Rake script that Kirk wrote (I think) and Vino was running when PARMS linkages needed to be moved and I added in the code to update the POB info in the Search records too. So PARMS linkages will have been moved too (you will notice references to Districts and Civil Parishes in the email logs).  
I’ll look into the Gazetteer tab issue.

**Comment by geoffj-FUG** [comment]
Anne

I have put PARMS Entry after some place names where they are in the wrong county so they can be connected to the PARMS.

May I suggest that when we have a situation similar to Axbridge then there be a test for PARMS links and if they are there then the old place name has 'PARMS Entry' appended to it instead of being deleted.

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
I think we perhaps need to discuss further. If you add 'PARMS Entry' to the name then that will affect the main FreeCen Search as it will show in the list of Census places with that appended to the name! This is all very complicated!! The emailed log from the Review mode run indicates if the place is used in PARMS (District, Civil Parish, Piece). The Search Census Place dropdown gathers names from freecen2_places (ie Gazetteer places) that are linked to Pieces with data loaded/incorporated.
As I said I based it on the existing code for moving linkages. When I have annotated the email logs for your test cases (if you could send those to me please) and you have had a chance to review we could discuss requirements again.

**Comment by geoffj-FUG** [comment]
Anne

 

Nobody should see these. Searches wont link as the places are in the wrong county but needed to make the PARMS structure work. We just flag the entries so that transcribers and validators don’t use them.

 

Geoff

**Comment by AnneV-Learn** [comment]
Your **Comment**
_When I run the review it is taking over my Gazetteer tab instead of appearing in my FreeCEN tab. So I end up with no gazetteer and 2 FreeCEN tabs._

Hopefully this won't happen after the code deployed to Test3 that made the Gazetteer open in the same tab by default has been revoked.

**Comment by geoffj-FUG** [comment]
Objective - to test whether the move affects the ability to upload using the PARMS.
Test 1 vld file to be uploaded to vld system.  Vld file belongs to Axbridge.
Step 1 move all Axbridge links to Dummy place name Backwell Voyseyville.
Review move - Proxy server timed out. Couldn't go back. Spinner so had to start at County Options (Breadcrumb) again. Move missing from menu options.
Reset at Your Actions. Goy to FreeCEN2 places and Move still missing.
Closed and reentered test 3. Still no Move
Logged out. Signed back in. Still no move.
Logged out, closed tab. Signed back in on new tab (Google) - still no move option.
Closed Google and restarted.  No move.
Checked syndicate and Role set at data manager.
Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG there must be something wrong with Test3 at the moment because I can’t even login to Test3 (Proxy server error). I’ll have to get Vino to look into it after the weekend (long weekend here in uk so that will be Tuesday).

**Comment by AnneV-Learn** [comment]
@geoffj-FUG I found a bit of code that was no doubt contributing to the slow running (-> PROXY server disconnect). This has been removed and Vino was around so she kindly deployed the update to Test3.
I am now able to connect to Test3, not sure why I couldn't connect earlier.

Regarding you not seeing the 'MOVE Linkages' button when connected to Test3 I have no idea about that - can only think you were somehow connect to Production!!!! When I am connected to Test3 it always shows **test3.freecen.org.uk** as the URL at the top of the screen, I use Safari, not sure what Web browser you use.

If you have the stamina you might want to test again - sorry this has all been so torturous.

**Comment by geoffj-FUG** [comment]
Anne

 

I am in trouble here.

 

I have rebooted the computer, used Google with a different Google user attached to it (FreeCEN sees it as a different person). Still no move menu item.

 

Given that Vino did an update the problem may have happened during the update. 

 

Does data manager still have permissions?

 

Geoff

**Comment by AnneV-Learn** [comment]
Ok Geoff I’ll take a look.
Anne

On 4 May 2024, at 11:47, Geoff J ***@***.***> wrote:

﻿

Anne



I am in trouble here.



I have rebooted the computer, used Google with a different Google user attached to it (FreeCEN sees it as a different person). Still no move menu item.



Given that Vino did an update the problem may have happened during the update.



Does data manager still have permissions?



Geoff

**Comment by geoffj-FUG** [comment]
Proxy issue has disappeared. It is very quick now.
Objective - to test whether the move affects the ability to upload using the PARMS.
Test 1 vld file to be uploaded to vld system. Vld file belongs to Axbridge.
Step 1 move all Axbridge links to Dummy place name Backwell Voyseyville.
Review move - report OK 
Moved Axbridge to Voyseyville all entries moved - year 1851, 12503 records 24 files.
Uploaded vld file RG121921. - 
Went to locate files and attempted to display RG12_1921 - got red message.
Attempted to locate RG12_1922 got red message.
Went to Manage vld files and scrolled to RG121921 - 7032 entries, 0 individuals, 0 swellings, upload details shown, Upload report says no errors.

In other words the changes to the parish have affected the PARMS and the file has not uploaded properly.

Recreated Axbridge in Gazetteer. Locate file RG12_1921, file details now appear.

Replace file RG121921.vld. Now shows 7032 entries, 6923 individuals, 1530 dwellings.

Conclusion - We have to retain the place in the Gazetteer and its links if the place has links to the PARMS. I have left the PARMS links pointing to Backwell Voyseyville so that you can examine the data if you need to.

Geoff

**Comment by geoffj-FUG** [comment]
Objectieve - t test the result when we move a place name search links to its alternative name.

Move Adelaide (currently the main entry, to South Australia Adelaide (currently the alternative name.
Note the results from the previous test are in my Gazetteer window. We need to manage this issue.
I do not have OVF in my list of counties to manage. 
(It is available in the counties in the live system. Note: OVB is also there, maybe we should remove the OVB option.)

Test stopped

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG I am going to respond to your first comment from 05 May above 
re Objective - to test whether the move affects the ability to upload using the PARMS vis email as there is too much detail to comment here.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG re 'Objectieve - t test the result when we move a place name search links to its alternative name comment above.'

Regarding the 'Results' from the previous test being in your Gazetteer window. When moving Place Linkage please do not have a separate Tab or Window open with the Gazetteer - make sure that is the case before Moving the Linkages in REVIEW or UPDATE mode. Having one open won’t do any actual damage but it will make it very confusing. 

OVF not being in your list of Counties to Manage on Test3, I am not at all familiar with the way that assigning Counties to manage works and really don’t want to mess things up. Could you please ask Vino to sort that for you?  

OVB on Live system - again not entirely sure of the consequences of removing that - or exactly how to do it. Again perhaps Vino could take that one. Probably best to raise a new Story.  

Moving a Place Name to its Alternative name. Hopefully this will be a rare thing to happen. I believe the way to accomplish this is as follows:
1. Create a temporary Place eg for your example 'Adelaide Temporary Place' with no Alternate Place names specified and perhaps an appropriate note indicating that it should not be used.
2. Take a note of the Alternative Place Names that you will want to retain.
3. Move Place Linkages for the Old Place (e.g. Adelaide). Note: In the UPDATE Mode run log it should actually report a list of the Alternative Names for the Old Place that are used in Search Records. Message is 'Old Place Alternate Place Names used in Search Records as Place of Birth =……..'
4. Still do NOT add any Alternative names to the temporary Place (ie Adelaide Temporary Place).
5. Ask System Administrator to Delete the Old Place (I.e. Adelaide). Note:This needs be done in the Mongo Database directly rather than through the application code.
6. Create the new place eg South Australia Adelaide - you can add the appropriate Alternative names now.
7. Move Place linkages for the temporary Place (ie Adelaide Temporary Place) to the new place (ie South Australia Adelaide)
  Perhaps this particular type of linkage move is best left to System Administrator completely.

**Comment by geoffj-FUG** [comment]
Objective
To move the search records from an alternative place name and make it the main place name.
SOM Spaxton has an alternative place name of Bridgwater Spaxton. So the objective is to make Bridgwater Spaxton the place name and Spaxton the alternative place name.
Step 1 - Create 2 place names in GLS called Test One and Test Two. This will mean that the data has to move across counties as well.

Problem - I do not have SOM Brdgwater Spaxton available to select in the specific places.

Revised objective:
To move the data from North Petherton and see the impact on Bridgwater North Petherton.
Number of records online for North Petherton 9391 23 files
Create alternative place name of Bridgwater North Petherton. OK
Select North Petherton as moving from OK
Select test 1 as move to place. Review results. Years include 1881, 1891.
Report received OK and details correct.
Move initiated. Reort OK.
Breadcrumbs have changed at top of Gazetteer page. 
Back to search page. Searched for North Petherton and it has disapperaed but so has the alternative name.
All records are currently stored correctly under GLS Test One

Question. Are search records stored fo the alternative name or are they included in the mainn place name search records (i.e one search code to cover all alternatives)?

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 

**Problem - I do not have SOM Brdgwater Spaxton available to select in the specific places.** => Only main place names are listing for selection as alternatives are intrinsically linked to the main place name (they are stored as an embedded array of the main place record). 
To make alterations to the alternative name you have to go via the main name. (I know it gets a bit complicated).

**Back to search page**.  => Do you mean then Gazetteer Search page here?

**Breadcrumbs have changed at top of Gazetteer page.**  => Could you please send a screenshot example of that - is it because on Test3 Gazetteer opens in same Tab?

**Searched for North Petherton and it has disappeared but so has the alternative name.** => When the move linkages rake is run in UPDATE mode it checks to see if any of the Alternative names for the OLD Place are currently used as POBs. If so they are listed towards the end of the log report, so that they can be added into the New Place name by the user after the run has completed. If they are not used then they are not listed as there is actually no need to add them in! ' Bridgwater North Petherton' is was not used anywhere as it had just been created by you.

**Comment by geoffj-FUG** [comment]
Anne

 

Have a look at story #1744 Turkey Smyrna that I have just put on the Board. It is by no means a unique, nor a rare, type of request.

 

Geoff

**Comment by geoffj-FUG** [comment]
Anne

 

Back to search page – Yes

 

Breadcrumbs. I will keep screen shots next time and send them to you. 

 

Validators add alternative names in the Gazetteer to help them locate a place when there are several similar places.

 

Have a look at Compton in SOM. There are several. I have added the greater nearby place to each entry so that when I am validating I can see whether my choice is likely or less likely. Otherwise I have to geolocate every time I get a Compton POB enumerated.

 

It would only be coincidental if there are search records attached to it.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
Ok so should all alternative names from the Old  place be added to the New place every time when running in UPDATE MODE (unless an old alternative matches the New place name)?

**Comment by geoffj-FUG** [comment]
Not all

 

We need to get rid of entries that are not Greater to Smaller, but keep others that are correct.

 

Geoff

 

PS closing now as it is getting late.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG
How about the move linkages process automatically adds alternative names that are recorded against the Old Place to the New Place if they were used as a POB (and report a this in the log) and just reports in the log any other alternative names from the Old Place so that the user can add them to the New Place if appropriate after the UPDATE mode run.

**Comment by geoffj-FUG** [comment]
Anne

 

I have had a think about this, and I believe that we need to amend several different actions:

*	The action that we are developing now, which is working well. Yes, When we move the main entry, it should move all the alternative entries as well.
*	The second action is an action that will delete an alternative place name. This can be the action menu item that we currently have for the Gazetteer. It should be enhanced so that when a search record or a PARMS link is there, these links and records are moved to the main place name and the alternative place name deleted.
*	Edit Alternative place name – Amend. When this is selected the new entry should be asked for. Then the system should create the new entry, attach any search / PARMS records and then delete the original entry. It will still be displayed as an alternative place name.

 

When the Edit is selected against a Gazetteer entry, we currently have a series of menu actions:

Add Alternative place name – Keep, no change.

Add Notes - Keep, no change.

Add source – Keep, no change.

Add website – Keep, no change.

Edit alternative place name – Amend, see above.

Edit location – Keep, no change.

Edit Place name – This applies to the main place name. It can no longer be done. The action should be removed from the list. It has been replaced by the new action being developed for Data Manager corrections.

Edir source – Keep, no change.

Edit website – Keep, no change.

 

I think (fingers crossed) that this should enable all the changes that we need to be made.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
Ok I’ll have to give this a lot more thought as the newly specified edit and delete actions would be heavy processing wise so not suitable to be run interactively but would entail initiating background Rake tasks.

**Comment by AnneV-Learn** [comment]
@Geoff-FUG Sorry but this is a lengthy and rather complex reply. 
Some info first:
Because changes to the underlying code related to Search Record creation by VLD uploads and CSVPro incorporations, and Search Record modification by the VLD POB validation  have been implemented (primarily to facilitate future Search by POB 'nearby places'). Search Records now store the POB and **internal linking id to the freecen2_place** (as long as it is a valid place for the birth county in the Gazetteer (freecen2_place). FYI - * I have almost completed the retrospective population of the internal linking id for all existing data, just one county to do now. 
Alternative place names are just held as embedded arrays of the main place in the freecen2_place record (Gazetteer) so any POB that is actually valid as a result of alternative name matching rather main place name matching have exactly the same internal linking id in the Search Record. 
BTW - Search Records hold all the searchable terms/dates etc. related to the source VLD/CSVPro transcription data record. 

All that means that I think we can handle your latest enhancement requirements as follows:

1. Move Place Linkages - yes I can modify the code to add all existing alternative places from the OLD place to the NEW place 
2. Destroy(Delete) Alternative place - any Search Records that have a POB that matched an alternative place will now (well after the retrospective population is completed - see * above) already have the correct internal id (i.e. that of the main place). So the Search Records do not need amending. 
 I am assuming the underlying VLD/CSVPro transcription data records should not be altered - need your thoughts on that.  
Also PARMS are linked via the internal id in the same way so should not need amending. 
3. Edit Alternative place name - following on from 2. above it should now be ok to edit the place name as the main place name internal id is as detailed above. Again Search Records/PARMS do not need amending.  
But the same question about whether the underlying VLD/CSVPro transcription data records should be altered at all in this case? 
4. The 'menu options' on Edit Place screen are there just to record the reason for making the change and do not control process flow. (Edits are all audit trailed). However Edit Place Name does need to be removed as you point out. 
5. If above strategy is agreed then the check to stop Alternative names being Destroyed/Deleted if they are used in Search Records can in fact be removed. Prior to the storing of the internal linking id to the freecen2_place for the POB the check was required because there was no route back to the freecen2_place for an alternative place name POB on the Search Record.   

**NOTE : None of the above can be deployed to Production until I have completed the retrospective population of linking id for POB on the Search Records.**

**Comment by geoffj-FUG** [comment]
Anne

 

My comments:

1                    Yes please

2                    OK – So the data manager can remove a place name. Can the Coordinator be allowed to do that as well? The underlying records do not need to be amended. They can stay as they are. PARMS are also OK.

3                    OK, the data manager can edit an alternative place name. The coordinator should be able to do that as well.

4                    Agree.

5                    If the check is no longer required, then we can allow the coordinator or validator to delete place names. So in that case the answer to 2 and 3 is yes.

This all sounds a bit simpler to what we had – certainly from the perspective of the coordinator and validator who are the people who make the amendments to the gazetteer.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG OK I've made the changes as per your last comments and they have been deployed to **Test3**.

**Comment by geoffj-FUG** [comment]
Objective - to test the change in search records counts when a new piece is incorporated containing records with the alternative place name as an entry. Then to move a place with an alternative place name to the main entry and see if the search record count changes and whether the alternative place name appears against the amalgamated gazetteer record.

RG14_14114. Created 8 entries with verbatim birth place of Minehead Higher Town. Minehead Higher Town is an alternative place name for Higher Town and is recorded as an alternative place name in the Gazetteer.  
Before I started there were no search records recorded for Higher Town.
RG14_14114 was incorporated and the count in the Gazetteer examined. Despite incorporating 8 records the count is still 0.

Problem 1 - why did the search record count not change when I incorporated the piece?
Problem 2 - before I incorporated this piece there were 17 files against Minehead. This piece would have been the 18th as there were many Minehead birth place entries as well as the 8 test records. However the filecount remains at 17.

Geoff

**Comment by geoffj-FUG** [comment]
Objective - To check that the number of search records does not change when an alternative place name is destroyed.

Walcot has an alternative place name  of Bath Walcot. Both Walcot and Bath Walcot appear regularly as place names.
Before the change there were 14829 search records in 45 pieces all in 1851.
After the change the count was the same.

Test OK. (Note that Bath Walcot has been re-entered into teh gazetteer).

Geoff

**Comment by geoffj-FUG** [comment]
Objective  - To transfer and amalgamate the search records for one place with another place which has its own set of search records.

The files for Dunster on test 3 covered the years 1861 to 1891. A 1911 piece RG14_14124 was incorporated to ensure that at least one csv file was included in the test.
Dunster Record count 16291 (after extra piece incorporated)
Files 16 (should be 17 as an additional piece was added) Possible Problem?

Old Cleeve is near to Dunster and Minehead. The search record count is 0 and the number of pieces is 0. The search record count should be more than 0 as there are records for Old Cleeve in the 1911 piece. Given their proximity Old Cleeve inevitably appears in pieces around Minehead / Dunster area and there should be a search record count given the number of files in the sample.

Problem - why are so many Gazetteer entries in test3 showing a search record count of 0? I have had a look at places like Bridgwater and Taunton and they have the same phenomenon).

Geoff

**Comment by geoffj-FUG** [comment]
There is another need - To combine 2 valid place names.

There are many cases where major towns have had their parishes scattered over several entries. All the place names are valid. It is therefore necessary to make some of these place names alternative place names of the primary place instead of main place names. We therefore need to option to move an entry to an alternative of another entry.
Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 

**Re: comments above relating to testing involving RG14_14114:**

_Problem 1 - why did the search record count not change when I incorporated the piece?_
'Number of search records online for this place' for a Gazetteer Place is the number of Search records with that place name as the **Census Place**. The Show Gazetteer Place screen does not include the number pf Search records with that place as POB. 

_Problem 2 - before I incorporated this piece there were 17 files against Minehead. This piece would have been the 18th as there were many Minehead birth place entries as well as the 8 test records. However the filecount remains at 17._
**17** is the Total Number of Pieces for the  Gazetteer Place Minehead not the number with VLD/CSV data loaded/incorporated.

The Pieces are as follows:
"HO107_1920" - No VLD files - No incorporated CSV files
"RG9_1601" - 1 VLD file (RG091601.VLD) - No incorporated CSV files
"RG10_2348" - No VLD files - No incorporated CSV files
"RG10_2349" - 1 VLD file (RG102349.VLD) - No incorporated CSV files 
"RG11_2353" - 1 VLD file (RG112353.VLD)  - No incorporated CSV files 
"RG12_1863" - 1 VLD file (RG121863.VLD) - No incorporated CSV files 
"RG13_2260" - No VLD files - No incorporated CSV files
"RG13_2261" - No VLD files - No incorporated CSV files
"RG14_14114" - No VLD files -  1 incorporated CSV file (rg14_14114.csv)
"RG14_14115" - No VLD files - No incorporated CSV files
"RG14_14118" - No VLD files - No incorporated CSV files
"RG14_14117" - No VLD files - No incorporated CSV files 
"RG14_14116" - No VLD files - No incorporated CSV files 
"RG14_14119" - No VLD files - No incorporated CSV files 
"RG14_14120" - No VLD files - No incorporated CSV files
"RG14_14121" - No VLD files - No incorporated CSV files
"RG14_14122" - No VLD files - No incorporated CSV files

**17** Pieces in Total

I then downloaded RG14_14115.CSV  from Production database - reduced number of records to just first 42. Loaded into Test3 and incorporated. Piece RG14_14115 now shows as  No VLD files -  1 incorporated CSV file (RG14_14115.CSV.csv = **Correct**.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 

**RE: comments above relating to Objective - To transfer and amalgamate the search records for one place with another place which has its own set of search records.**

_The files for Dunster on test 3 covered the years 1861 to 1891. A 1911 piece RG14_14124 was incorporated to ensure that at least one csv file was included in the test. Dunster Record count 16291 (after extra piece incorporated) Files 16 (should be 17 as an additional piece was added) Possible Problem?_

**16** is the number of Pieces for the Gazetteer place Dunster not the number of VLD/CSV Files loaded/incorporated.

_Old Cleeve is near to Dunster and Minehead. The search record count is 0 and the number of pieces is 0. The search record count should be more than 0 as there are records for Old Cleeve in the 1911 piece. Given their proximity Old Cleeve inevitably appears in pieces around Minehead / Dunster area and there should be a search record count given the number of files in the sample._
_Problem - why are so many Gazetteer entries in test3 showing a search record count of 0? I have had a look at places like Bridgwater and Taunton and they have the same phenomenon)._

The Number of search records online for a Gazetteer place is the number of Search Records that have been loaded/incorporated for the Place via Pieces that are 'for' the place ie a Census Place. Old Cleeve is not a Census Place (ie it has no Pieces directly linked to it). The Show Gazetteer Place screen info does not include the number of Search records with that place as POB. 
 When I checked Bridgwater it has 47,157 Search records and 39 Pieces. Taunton (SOM) has 0 Search Records, 22 Pieces and Data present = false. These look ok to me.

BTW - Data present for this place: is true if number of Search Records is > 0.

So summing up, the info in the Gazetteer (Show) only relates to the Place as a 'Census' Place and does not show any Stats relating to POBs. This is because the full linkage as not there for POBs. I have Completed retrospective population of the full linkage in the Production database, so that for every Search record that has a valid POB (Alternative if not Verbatim) the full link to the Gazetteer place is stored. The VLD load and CSV incorporate processes also now populate this link as part of their processing, as does the VLD POB validation process. The same retrospective population has not been done for TEST3 (it takes a long time!).
I was thinking that it would be a good idea if we 'replaced' the existing TEST3 database with a copy of the Production database at some point in the POB Searching development work as TEST3 has a number of anomalies which can make testing tricky.

To add the POB info the the Gazetteer info for a Place was not possible before due to the amount of processing involved as no full linkages and even now that we have the linkages I am not entirely sure that it would be feasible. We would certainly have to add some additional indexing to the Search records (which we would no doubt need to do for full implementation of POB searching anyway). Having a True/False flag for POB data present for this place and Changing the 'names' for the Place info as follows might help as things stand perhaps with a note that it does not indicate if the data has been used as a POB..  
Data present for this place: -> Data present for this Census place
Census years present for this place -> Census years present for this Census Place Number of search records online for this place -> Number of search records online for this Census Place  (Census Place being the term used on the Search page.)

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 

**RE: There is another need - To combine 2 valid place names.**

If I am understanding the requirement correctly then this is already achievable as follows:
Move the Secondary Place Linkages (OLD Place) to the Primary Place (NEW Place) - Via the 'Move Freecen2 Place' button and after that has been successfully run in UPDATE mode you will able to add the Secondary Place name (OLD Place name keep a note of it!) as an Alternative Name to the updated Primary Place (NEW Place). Hope that makes sense.

**Comment by geoffj-FUG** [comment]
Anne

 

Yes, that makes sense.

 

Geoff

**Comment by geoffj-FUG** [comment]
OK. So I have been looking for more thann is available.

This can therefore be deployed as I have effectively tested every option.
My queries are explained.

On deployment we need to remove the following Edit action:
Edit Place name.
This is now an action to be taken by the Data manager using  the new code.
Once live I will test the system using some of the stories that I have put on ice.

Geoff


Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
I need to complete correction of place ids on Search records with birthplace = hyphen #1788 before this can be deployed. 

Also I’ll add in the removal of the ‘Edit Place Name’ to the code for this story so that it all be deployed at once.

**Comment by AnneV-Learn** [comment]
Correction of place ids on Search records with birthplace = hyphen (#1788) has now been completed, so this can be deployed to Production.

**Comment by Vino-S** [comment]
Changes deployed to production.
